pkg/asset/installconfig/baremetal/host.go

```diff
@@ -0,0 +1,182 @@
+		Help:    "Root Device Hints are for specifying the storage location for the root filesystem for the image.",
+	}, &rdHints, nil)
+	if rdHints {
+		var rootDeviceHints v1alpha1.RootDeviceHints
+		for {
+			var hint string
+			survey.AskOne(&survey.Select{
+				Message: "Select Hint:",
+				Options: []string{"DeviceName", "HCTL", "Model", "Vendor", "SerialNumber", "MinSizeGigabytes", "WWN", "WWNWithExtension", "WWNVendorExtension", "Rotational"},
+			}, &hint, nil)
+			gatherRootDeviceHints(&rootDeviceHints, hint)
+
+			more := false
+			survey.AskOne(&survey.Confirm{
+				Message: "Specify more hints?",
+			}, &more, nil)
+			if !more {
+				break
+			}
+		}
+		host.RootDeviceHints = &rootDeviceHints
+	}
+
+	return &host, nil
+}
+
+func gatherRootDeviceHints(rootDeviceHints *v1alpha1.RootDeviceHints, hint string) error {
+	hintHelpMessage := map[string]string{
+		"DeviceName":         "A Linux device name like '/dev/sda'.",
+		"HCL":                "A SCSI bus address like 0:0:0:0.",
+		"Model":              "A vendor-specific device identifier.",
+		"Vendor":             "The name of the vendor or manufacturer of the device.",
+		"SerialNumber":       "A vendor-specific device identifier.",
+		"MinSizeGigabytes":   "Device serial number.",
+		"WWN":                "Unique storage identifier.",
+		"WWNWithExtension":   "Unique storage identifier with vendor extention appended.",
+		"WWNVendorExtension": "Unique vendor storage identifier.",
+		"Rotational":         "True if the device should use spinning media, false otherwise.",
+	}
+	answer := ""
+	// MinSizeGigabytes is a number
+	if hint == "MinSizeGigabytes" {
+		survey.AskOne(&survey.Input{
+			Message: hint,
+			Help:    hintHelpMessage[hint],
+		}, &answer, isNumber)
+		sizeInt, err := strconv.Atoi(answer)
+		rootDeviceHints.MinSizeGigabytes = sizeInt
+		return err
+	}
+	// Rotational is a boolean
+	if hint == "Rotational" {
+		rotational := true // default value in the platform
+		survey.AskOne(&survey.Confirm{
+			Message: hint,
+		}, &rotational, nil)
+		rootDeviceHints.Rotational = &rotational
+		return nil
+	}
+	// All other hints are strings
+	survey.AskOne(&survey.Input{
+		Message: hint,
+		Help:    hintHelpMessage[hint],
+	}, &answer, isNotEmpty)
+	switch hint {
+	case "DeviceName":
+		rootDeviceHints.DeviceName = answer
+	case "HCTL":
+		rootDeviceHints.HCTL = answer
+	case "Model":
+		rootDeviceHints.Model = answer
+	case "Vendor":
+		rootDeviceHints.Vendor = answer
+	case "SerialNumber":
+		rootDeviceHints.SerialNumber = answer
+	case "WWN":
+		rootDeviceHints.WWN = answer
+	case "WWNWithExtension":
+		rootDeviceHints.WWNWithExtension = answer
+	case "WWNVendorExtension":
+		rootDeviceHints.WWNVendorExtension = answer
+	}
+
+	return nil
+}
+
+// isNumber validates if the answer provided is a valid number
+func isNumber(ans interface{}) error {
+	_, err := strconv.Atoi(ans.(string))
+	return err
+}
+
+// isNotEmpty validates if the answer provided is a valid number
```

Comment:
the documentation seems like a copy paste error.

Reply:
+1. Probably moot, if we remove root device hints anyways.

Reply:
Removed it. No longer necessary.